spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -81,6 +83,10 @@
 
 	private volatile AbstractMessageChannelMetrics channelMetrics = new DefaultMessageChannelMetrics();
 
+	public AbstractMessageChannel() {
+		this.interceptors = new ChannelInterceptorList(logger);
+	}
```

Comment:
I think you should agree that this is legit:
```
private final Foo foo = new Foo();

private final Bar bar = new Bar(this.foo);
```
Because of an order of field initialization.
However I won't mind for that ctor too. Just to be sure :smile: